src/sparse_merkle_tree/smt.rs

```diff
@@ -1362,24 +1364,31 @@ impl SparseMerkleTree {
 
     pub fn prepare_queries_with_proof_map(
         proof: &Proof,
-    ) -> HashMap<Vec<bool>, QueryProofWithProof> {
+    ) -> Result<HashMap<Vec<bool>, QueryProofWithProof>, SMTError> {
         let mut queries_with_proof: HashMap<Vec<bool>, QueryProofWithProof> = HashMap::new();
-        for query in &proof.queries {
-            let binary_bitmap = utils::strip_left_false(&utils::bytes_to_bools(&query.bitmap));
-            let binary_path = utils::bytes_to_bools(&query.pair.0)[..binary_bitmap.len()].to_vec();
-
-            queries_with_proof.insert(
-                binary_path,
-                QueryProofWithProof::new_with_pair(
-                    Arc::clone(&query.pair),
-                    &binary_bitmap,
-                    &[],
-                    &[],
-                ),
-            );
+        if !proof.queries.is_empty() {
+            let max_binary_bitmap_len = proof.queries[0].key().len() * 8; // converts key_length to bits
```

Comment:
Is this really maximum bitmap length?
Can some other query has longer key? (eg. `proof.queries[15]`)

Reply:
The length of each key in queries are same also it is already check with `key_length` in `verify` function line `1550`